OpenRA.Mods.Common/Traits/Buildings/BuildingUtils.cs

```diff
@@ -17,15 +17,25 @@ namespace OpenRA.Mods.Common.Traits
 {
 	public static class BuildingUtils
 	{
-		public static bool IsCellBuildable(this World world, CPos cell, BuildingInfo bi, Actor toIgnore = null)
+		public static bool IsCellBuildable(this World world, CPos cell, ActorInfo ai, BuildingInfo bi, Actor toIgnore = null)
 		{
 			if (!world.Map.Contains(cell))
 				return false;
 
+			// cell is occupied by a structure
 			if (world.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(cell) != null)
-				return false;
-
-			if (!bi.AllowInvalidPlacement && world.ActorMap.GetActorsAt(cell).Any(a => a != toIgnore))
+			{
+				// check for 'Replacement'-trait & if it fits
+				if (ai == null)
+					return false;
+
+				var replacementInfo = ai.TraitInfoOrDefault<ReplacementInfo>();
+				if (!(replacementInfo != null &&
+					IsBuildingAtCellReplaceable(world, cell, replacementInfo)))
+					return false;
+			}
+			else
```

Comment:
The else if should be on one line, please

Reply:
Will change it.
I was not 100 percent sure, if i've handled the if-statement right.